test/aspect/aspect.bzl

```diff
@@ -1,49 +1,56 @@
-attr_aspects = [
-    "deps",
-    "runtime_deps",
-    "_scalalib",
-    "_scalareflect",
-    "_scalatest_reporter",
-]
+"""
+This test makes sure that the implicit rule dependencies are discoverable by
+an IDE. We stuff all dependencies into _scala_toolchain so we just need to make
+sure the targets we expect are there.
+"""
+scala_toolchain = "_scala_toolchain"
 
 def _aspect_impl(target, ctx):
-    visited = [target.label.name]
-    for name in attr_aspects:
-        if hasattr(ctx.rule.attr, name):
-            attr = getattr(ctx.rule.attr, name)
-            # Need to handle whether attribute is label or label list
-            children = attr if type(attr) == "list" else [attr]
-            for child in children:
-                if hasattr(child, "visited"):
-                    visited += child.visited
+    visited = [str(target.label)]
+    if hasattr(ctx.rule.attr, scala_toolchain):
+        for dep in getattr(ctx.rule.attr, scala_toolchain):
+            if hasattr(dep, "visited"):
+                visited += dep.visited
     return struct(visited = visited)
 
 test_aspect = aspect(
-    attr_aspects = attr_aspects,
+    attr_aspects = [scala_toolchain],
     implementation = _aspect_impl,
 )
 
 def _rule_impl(ctx):
-    expected = [
-        "dummy",
-        "jar", # This is scalatest since @scalatest/jar
-        "scala-library",
-        "scala-reflect",
-        "scala-xml", # dependency of test_reporter
-        "test_reporter",
-    ]
-    # Remove duplicates and sort so can do simple comparison
-    visited = sorted(depset(ctx.attr.scala_rule.visited).to_list())
-    if visited == expected:
-        content = "true"
-    else:
-        content = """
-        echo Expected these rules to be visited by the aspect: 1>&2
-        echo %s, 1>&2
-        echo but got these instead: 1>&2
-        echo %s 1>&2
-        false
-        """ % (', '.join(expected), ', '.join(visited))
+    deps = {
```

Comment:
It's used below as: `expected = deps[rule.label.name]`
Do you just not like the name?